### PR TITLE
Revamped the pie chart widget

### PR DESCRIPTION
**Pull request by IanWoodard**
Revamped the existing pie chart widget to bring it closer to the new spec defined in Figma. I do want to mention that although the spec does not define any hover state for the pie chart widget. If this is because there is not meant to be any, let me know and I can remove that from the PR. I wasn't sure if the hover/highlight interaction was being left out, but since the existing logic was already there and because I thought it looked really cool, I edited the logic to simplify it a bit and made some stylistic changes to make it better fit the new spec. Once again, let me know if you want me to leave these changes out and or if you want me to make adjustments to them.

Additionally, I am unsure how to get the APR/APY so I left placeholders for those values for the time being.

As usual, I added some images of the updated component below:

### Default:
revamped_pie_chart_default

### Active 1:
revamped_pie_chart_active_1

### Active 2:
revamped_pie_chart_active_2

### CODE REVIEW

**Review by haydenshively — APPROVED** [review]

File: src/components/poolstats/PoolPieChartWidget.tsx [hunk 1]
```diff
@@ -33,68 +25,20 @@ const PieChartContainer = styled.div`
...
-    mouseData: state,
-    handleMouseMove,
-    handleMouseLeave,
-  };
-};
-
 // MARK: Pie chart setup -------------------------------------------------------------
 
 type PieChartSlice = {
+  index: number;
```
Comment (haydenshively) [inline src/components/poolstats/PoolPieChartWidget.tsx:31]: `index` as a property of `PieChartSlice` makes sense, as it's general purpose and makes it possible to remove the mouseMoveData atan2 stuff. Nice work

File: src/components/poolstats/PoolPieChartWidget.tsx [hunk 2]
```diff
@@ -33,68 +25,20 @@ const PieChartContainer = styled.div`
...
   percent: number;
   color: string;
+  category: string;
+  metric?: string;
```
Comment (haydenshively) [inline src/components/poolstats/PoolPieChartWidget.tsx:38]: However, I'm not sure why category and metric should be part of this type. If we were to create another pie chart elsewhere, would these things stick around, or are they specific to this use-case? If specific to this use-case, I think we should `extend` `PieChartSlice` to something like `AllocationPieChartSlice` and put the extra props there

File: src/components/poolstats/PoolPieChartWidget.tsx [hunk 3]
```diff
@@ -142,38 +82,143 @@ const PieChartLabel = styled.div.attrs<{ shouldShow: boolean }>((props) => {
...
     height: 8px;
     aspect-ratio: 1;
     margin-top: -4px;
 
     border-radius: 50%;
     background: ${({ color }) => color};
+    z-index: 5;
+  }
+
+  :first-child:after {
```
Comment (haydenshively) [inline src/components/poolstats/PoolPieChartWidget.tsx:135]: Can you add comments to explain what all these child element selectors are doing?

File: src/components/poolstats/PoolPieChartWidget.tsx [hunk 4]
```diff
@@ -255,110 +303,121 @@ export default function PoolPieChartWidget(props: PoolStatsWidgetProps) {
       : ` and ${drawData.silo1Label}`
   );
 
-  const mouseIsInsidePieChart = mouseData.isActive && mouseData.r < 1;
+  const firstHalfOfSlices = slices.slice(0, slices.length / 2).reverse();
```
Comment (haydenshively) [inline src/components/poolstats/PoolPieChartWidget.tsx:314]: Is this split between first half and second half of slices really necessary?

**Review by haydenshively — CHANGES_REQUESTED** [review]
Feels like you removed a lot of complexity by using the indices instead of `atan2` with mouse position (for hovering). (Yay!) But then you added more complexity with the first half / second half of slices logic. Is this really necessary?

File: src/components/poolstats/PoolPieChartWidget.tsx (hunk 4 above)
Comment (IanWoodard) [reply src/components/poolstats/PoolPieChartWidget.tsx:314]: With the current implementation/simplification, yes. It is one of the tradeoffs made. The reason is that I want it to be able to support different numbers of slices, which hardcoding would not allow.

Comment (haydenshively) [reply src/components/poolstats/PoolPieChartWidget.tsx:314]: iirc it already supported different numbers of slices?

**Review by haydenshively — APPROVED** [review]
Approving since nits don't impact user-facing functionality, but may come back to this and build a generalized PieChart component in the future

**Pull request merged**